pkg/cloudprovider/types.go

```diff
@@ -82,8 +82,19 @@ type CapacityConstraints struct {
 	Architecture Architecture
 }
 
-// CapacityPacking is a solution to packing pods onto nodes given constraints.
-type CapacityPacking map[*v1.Node][]*v1.Pod
+// PodPackings is a solution to packing pods onto nodes given constraints.
+type PodPackings map[*v1.Node][]*v1.Pod
+
+type Packings struct {
+	Pods                []*v1.Pod
+	InstanceTypeOptions []string
+}
+
+// Packer is a method that takes contraints and calculates the pods that can be
+// efficiently placed on the instances.
+type Packer interface {
```

Comment:
Likewise -- Unless I'm misunderstanding something, this interface doesn't belong here.

I could imagine that we replace our Capacity.Create() interface with something similar to what you're going for here (e.g. Packer.Pack), but then we'd need to include constraints as well. I actually quite like the idea of separating pods from the constraints struct, so the cloud provider's implementation is effectively: `packer.Pack(ctx, pods, constraints)`